### DISCUSSION TITLE
Extracting Coverage map ray paths information

### DISCUSSION THREAD

**Discussion by Sina7474** [NVlabs/sionna#710] [Q&A]
Hello,

I am working with Sionna ray tracing to optimize RIS location deployment, focusing on coverage maps to avoid RIS limitations for multi-reflections. While using the Paths.CIR() function, I successfully extracted key ray information, including reflection points (Paths.vertices), interacting object IDs (Paths.objects), and ray power.

I would now like to extract similar data while performing Sionna coverage map computations. Could you please guide me on how to retrieve this information effectively?

Thank you very much for your support.

Best regards,
Sina

**Comment by faycalaa** [comment]
Hi @Sina7474,

Such information is not stored when computing coverage map. You could edit the code of the coverage map solver, and especially the [shoot-and-bounce main loop](https://github.com/NVlabs/sionna/blob/12d90787f5a67083ec6d31013405bb479cc7a619/sionna/rt/solver_cm.py#L2691), to extract  the information you need.
